### ISSUE TITLE
Collect non 200 responses

### ISSUE THREAD

**Issue by danielo515** [shoreditch-ops/artillery#190]
Hello,

I'm not sure if this is currently possible, but if it is, please instruct me how.
I want to see what uris failed with which statusCode, which for me is basically any status different from 200.

I'm currently trying with this:

```js
module.exports.logErrors = function logArtilleryErrors(requestParams, response, context, ee, next) {
    if (response.statusCode !== 200) {
        ee.emit('nonOkResponse', {uri: requestParams.uri, statusCode: response.statusCode})
    }
    next();
}
```

But nothing appears on the log. What I'm doing wrong?

Regards

**Comment by hassy** [comment]
Not possible at the moment. What I've done before myself is use `console.error()` and then redirect stderr to a file when running Artillery with something like `artillery run something.yaml 2>errors.txt`

This will eventually be solved properly - there's an open issue here: https://github.com/shoreditch-ops/artillery/issues/161

**Comment by danielo515** [comment]
I was fearing an answer like that.
Thanks for the workaround by the way.

> Logging histograms should be possible
I'm curious about that already implemented feature. What does it refers to?

**Comment by paraspatidar** [comment]
This is something which might help to those who are looking for similar thing :  
pre-requisite to log in file  :  `npm install -g fs`

testconfig.yaml -->

`config:
  target: 'http://192.168.1.2'
  phases:
    - duration: 1
      arrivalRate: 1
  defaults:
    headers:
      x-my-service-auth: '987401838271002188298567'
  processor: "my-functions.js"
scenarios:
  - flow:
    - get:
        url: "/fnempty01"
        afterResponse: "logHeaders"`


related my-functions.js in same folder as this config : 

`const fs = require('fs');

 module.exports = {
    setJSONBody: setJSONBody,
    logHeaders: logHeaders
  }

  function setJSONBody(requestParams, context, ee, next) {
    return next(); // MUST be called for the scenario to continue
  }

  function logHeaders(requestParams, response, context, ee, next) {
    console.log(response.body);
	let body=response.body;
	fs.writeFile('temp.txt', body, function(err, body){
    if (err) console.log(err);
    console.log("Successfully Written to File.");
	});
    return next(); // MUST be called for the scenario to continue
  }

  `


now just do 
`artillery run testconfig.yaml`

**Comment by hassy** [comment]
There's no need to `npm install` the `fs` package - it's a built in module in Node.js.

Also bear in mind that writing to a file for every response will slow down your tests unless the RPS is very low.

Otherwise, it's a decent solution e.g. if you're running a single virtual user to help debug why something might be failing.

**Comment by paraspatidar** [comment]
@hassy , Yeah thats exactly is requirement to verify if something is failing then for debug purpose.
as fiddler or any other tracing tool was unable to capture the same load test traffic which **artillery** is making.
Can you help on that one on how to capture same traffic on standard tools like fiddler / wire-shark.

Or Shall  I open any ticket for separate documentation ?

**Comment by hassy** [comment]
@paraspatidar there's nothing special about how Artillery sends its traffic, so using Wireshark as normal should just work. It would be good to have an example blog post perhaps, but having it as part of the documentation would be out-of-scope for that reason.
